### PR TITLE
Added CyberPanel password changing plugin

### PR DESCRIPTION
**Pull request by hifihedgehog**
In a recent security hotfix, CyberPanel introduced Bcrypt password encryption for all passwords. Previously, I maintained a tutorial online on how to change passwords with the custom SQL password changing plug-in. It was done easily enough since passwords at the time were stored in plain text format. However, now, since only a PHP function can perform the necessary encryption, this CyberPanel hotfix necessitated moving away from the custom SQL password changing plug-in to a full-on, custom-made plug-in specific to CyberPanel. So, using your development information online, I whipped one up! Feel free to add it to your project.

### CODE REVIEW

**Comment by hifihedgehog** [comment]
Pinging @usmannasir from CyberPanel for his stamp of approval.

File: plugins/change-password-cyberpanel/ChangePasswordCyberPanel.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		$bResult = false;
+		$db = mysqli_connect($this->mHost, $this->mUser, $this->mPass, 'cyberpanel');
+
+		try
+		{
+			$sEmail = $oAccount->Email();
+			$sEmailUser = \MailSo\Base\Utils::GetAccountNameFromEmail($sEmail);
+			$sEmailDomain = \MailSo\Base\Utils::GetDomainFromEmail($sEmail);
+
+			$password_check_query = "SELECT * FROM e_users WHERE emailOwner_id = '$sEmailDomain' AND email = '$sEmail'";
```
Comment (RainLoop) [inline plugins/change-password-cyberpanel/ChangePasswordCyberPanel.php]: I think here you need to escape special characters in the values.

Comment (hifihedgehog) [reply plugins/change-password-cyberpanel/ChangePasswordCyberPanel.php]: Thanks for the review. Looking closely, I don't immediately see an escape character pass in the custom SQL password plug-in which I drew heavily upon in drafting this. Do you have a recommendation or preference for the method you would like this to escape special characters? I could simply use the mysqli_real_escape_string() for pre-processing the in-line variables. Is that what you are after or something more case specific?

Comment (hifihedgehog) [reply plugins/change-password-cyberpanel/ChangePasswordCyberPanel.php]: I added mysqli_real_escape_string processing to all relevant variables that could potentially be used in the SQL statement.

**Comment by hifihedgehog** [comment]
Is there anything else you need completed by me @RainLoop before this will be accepted?

**Comment by usmannasir** [comment]
Is there anything I need to check on my end?

**Comment by hifihedgehog** [comment]
Please do. RainLoop asked me to escape for special characters which was a simple addition. You could test it out, most definitely, though. Maybe seeing that CyberPanel's lead dev is replying could help give them the push to integrate this. Understandably, they usually take some time to reply.

File: plugins/change-password-cyberpanel/ChangePasswordCyberPanel.php (hunk 1 above)
Comment (hifihedgehog) [reply plugins/change-password-cyberpanel/ChangePasswordCyberPanel.php]: Is there anything else you need me or CyberPanel's @usmannasir to do before you'll approve this pull request?

**Comment by hifihedgehog** [comment]
@usmannasir: I just added a separate repo that contains the plug-in alone and the initial 1.0 release.

**Pull request merged**